src/ui/lua/ScriptObject.mjs

```diff
@@ -0,0 +1,83 @@
+import {
+  LUA_REGISTRYINDEX,
+  lua_createtable,
+  lua_getglobal,
+  lua_pushlightuserdata,
+  lua_pushnumber,
+  lua_rawgeti,
+  lua_rawset,
+  lua_setglobal,
+  lua_setmetatable,
+  lua_settop,
+  lua_type,
+  luaL_ref,
+} from './api';
+
+const metaTables = new Map();
+const objectTypes = new Map();
+
+class LuaScriptObject {
```

Comment:
Let's stick to `FrameScriptObject` or `BaseScriptObject`.

Reply:
Renamed.